src/App.tsx

```diff
@@ -1,27 +1,70 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.scss';
+import 'bulma/css/bulma.css';
 import { GoodsList } from './GoodsList';
+import { Good } from './types/Good';
+import { getAll, get5First, getRedGoods } from './api/goods';
 
-// import { getAll, get5First, getRed } from './api/goods';
-// or
-// import * as goodsAPI from './api/goods';
+export const App: React.FC = () => {
+  const [goods, setGoods] = useState<Good[]>([]);
 
-export const App: React.FC = () => (
-  <div className="App">
-    <h1>Dynamic list of Goods</h1>
+  const handleLoadAllGoodsClick = () => {
+    getAll()
+      .then(AllGoods => setGoods(AllGoods));
+  };
 
-    <button type="button" data-cy="all-button">
-      Load all goods
-    </button>
+  const handleFirst5GoodsClick = async () => {
+    setGoods(await get5First());
+  };
 
-    <button type="button" data-cy="first-five-button">
-      Load 5 first goods
-    </button>
+  const handleRedGoodsClick = async () => {
+    setGoods(await getRedGoods());
+  };
```

Comment:
you can use one function instead of three, like this:
  const handleButtonSubmit = async (callback: Promise<Good[]>) => {
    setVisibleGoods(await callback);

and call it: onClick={() => handleButtonSubmit(getAll())}

Reply:
will be better use another name for callback))  like loadedGoods, for example